server/src/instant/db/instaql.clj

```diff
@@ -183,14 +183,15 @@
                             (grow-paths path)))})
 
         (and (map? v) (contains? v :$isNull) (= true (:$isNull v)))
-        ;; If the where cond has `$isNull=true`, then we
-        ;; need it should match if any of the intermediate
-        ;; paths are null
-        (let [path (string/split (name k) #"\.")]
-          {:or (concat [[[path v]]]
-                       (map (fn [p]
-                              [[p {:$isNull true}]])
-                            (grow-paths path)))})
+        ;; If the where cond has `$isNull=true`, then we need it to
+        ;; match if any of the intermediate paths are null
+        (let [path (string/split (name k) #"\.")
+              conds (map (fn [p]
+                           [[p {:$isNull true}]])
+                         (grow-paths path))]
+          (if (= 1 (count conds))
+            {:and conds}
+            {:or conds}))
```

Comment:
The changes here fix a bug where we would do an `or` with the same query twice.

Reply:
There's still a slowdown when using ors that I'm looking into, though.

Reply:
One hunch I always check: if there's any accidental cross join in the intermediate CTEs